notifications.api.service.ts

loretdemolas opened the pull request:
-created api call to delete notification and included a consol.log response to verify call.

notifications.model.service.ts
-add a function to pass data from logic page to api service

notifications.page.html
- modified onDeleteNotification to pass notifcation instead of notifcation.id to keep it in line with current style

notifications.page.ts
-Created a function to call the model service to delete function, -add a fuction to onDeletenotification to call model service init to update the notificaiton.

loretdemolas committed:
notifications.api.service.ts
-created api call to delete notification and included a consol.log response to verify call.

notifications.model.service.ts
-add a function to pass data from logic page to api service

notifications.page.html
- modified onDeleteNotification to pass notifcation instead of notifcation.id to keep it in line with current style

notifications.page.ts
-Created a function to call the model service to delete function,
-add a fuction to onDeletenotification to call model service init to update the notificaiton.

haxwell commented on src/app/pages/notifications-page/notifications.page.ts:
```diff
@@ -1,5 +1,6 @@
 import { Component, OnInit } from '@angular/core';
 import { NotificationModelService } from './_service/notifications.model.service';
+import { data } from 'cypress/types/jquery';
```
Don't think you meant to / need to include this import.

loretdemolas replied:
Fixed!

haxwell commented on src/app/pages/notifications-page/notifications.page.ts:
```diff
@@ -50,4 +51,11 @@ export class NotificationPage implements OnInit {
+
+  onDeleteNotification(notification: any) {
+    this.notificationModelService.deleteNotification(notification.id).then((data)=>{
+      this.notificationModelService.init();
```
notificationModelService.deleteNotification() should call .init() itself..

haxwell replied:
..to clarify, as opposed to having the component which called the service know to initialize the model, the service should know to do that, and it should do it itself. It will always need to be done after a delete, so the component should not be given that responsibility.

loretdemolas replied:
Would this change be what you mean?

loretdemolas committed:
notifications.model.service.ts
-moved function call from page.ts to model.service
notifications.page.ts
-removed bad import
-refactored function call to model.service
